v2/cli/init_command.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/brigadecore/brigade/v2/internal/file"
+	"github.com/urfave/cli/v2"
+)
+
+type initFields struct {
+	ProjectID   string
+	Language    string
+	GitCloneURL string
+	Script      string
+}
+
+var initCommand = &cli.Command{
+	Name:  "init",
+	Usage: "Initialize a new Brigade project",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    flagID,
+			Aliases: []string{"i"},
+			Usage: "Name of the Brigade project to initialize " +
+				"(required)",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLanguage,
+			Aliases: []string{"l"},
+			Usage: "Specify the scripting language to use for handling events" +
+				" -- JavaScript (js) or TypeScript (ts)",
+			Value: "ts",
+		},
+		&cli.StringFlag{
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage: "Optionally specify a git repository where the" +
+				" event handling script can be found",
+		},
+	},
+	Action: initialize,
+}
+
+// Handle init CLI flags
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := initFields{
+		ProjectID:   c.String(flagID),
+		Language:    strings.ToLower(c.String(flagLanguage)),
+		GitCloneURL: c.String(flagGit),
+	}
+
+	err := idMatch(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = canonicalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	if dirExists, dirErr := file.Exists(".brigade"); dirErr != nil {
+		return dirErr
+	} else if !dirExists {
+		fmt.Println("Creating .brigade directory...")
+		err = os.Mkdir(".brigade", 0755)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Generate js/ts buffer
+	scriptBuf, scriptErr :=
+		execTemplate(scriptTemplate, fields)
```

Comment:
I like that you're returning a byte array now. It's more straightforward. In light of this, the variable you assign that result to probably should no longer have "buf" in its name. Something like `scriptBytes` would make sense. This applies to other invocations of `execTemplate()` as well.

Reply:
It's out of the ordinary to not use/reuse simply `err` for errors. I suspect you chose this alternative name because the linter gave you grief about a "shadow declaration?" If so, I'll show you how to resolve that more cleanly.

Reply:
Yup exactly, I had to change the name of the error because of that reason!

Reply:
The short an answer to explicitly declare the _other_ variable separately above so that err can be reused without redeclaring (i.e. without the colon in the assignment). This stops you from making a new err variable with the same name as the old one, which has the potential to generate confusion-- which is why the linter complains.